Recrutify/Recrutify.Services/Services/ScheduleService.cs

```diff
@@ -81,6 +93,81 @@ public async Task UpdateScheduleSlotsForCurrentUserAsync(IEnumerable<DateTime> d
             await _scheduleRepository.BulkUpdateScheduleSlotsAsync(currentUserId, newListDateTime, removedListDateTime);
         }
 
+        public async Task ProcessingAppointOrCancelInterviewsAsync(IEnumerable<InterviewDTO> interviews, Guid projectId, CancellationToken cancellationToken)
+        {
+            var projectExists = await _projectService.ExistsAsync(projectId, cancellationToken);
+            if (!projectExists)
+            {
+                throw new Exception("Project does not exist.");
+            }
+
+            var allCadidates = await _candidateService.GetCandidatesByIdsAsync(interviews.Select(i => i.CandidateId));
+
+            var cancelledInterviews = _mapper.Map<IEnumerable<Interview>>(interviews.Where(i => i.IsAppointment == false));
+            var cadidatesIdsCancelledInterviews = cancelledInterviews.Select(i => i.CandidateId);
+            var cuurentStatusCadidates = allCadidates.Where(c => cadidatesIdsCancelledInterviews.Contains(c.Id)).Select(c => c.ProjectResults.FirstOrDefault(p => p.ProjectId == projectId).Status).FirstOrDefault();
+
+            if (cancelledInterviews.Any())
+            {
+                await _scheduleRepository.BulkСancelInterviewsAsync(cancelledInterviews);
+                await _candidateService.BulkUpdateStatusAsync(cancelledInterviews.Select(i => i.CandidateId), projectId, _statusHelper.GetStatusDown(cuurentStatusCadidates));
```

Comment:
in line 107 you've already done select ids

Reply:
+